mathjax3-ts/input/tex/base/BaseMethods.ts

```diff
@@ -0,0 +1,1625 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+/**
+ * @fileoverview The Basic Parse methods.
+ *
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+import * as sitem from './BaseItems.js';
+import {StackItem, EnvList} from '../StackItem.js';
+import {Symbol, Macro} from '../Symbol.js';
+import {ParseMethod} from '../Types.js';
+import NodeUtil from '../NodeUtil.js';
+import TexError from '../TexError.js';
+import TexParser from '../TexParser.js';
+import {TexConstant} from '../TexConstants.js';
+import ParseUtil from '../ParseUtil.js';
+import {MmlNode, TEXCLASS} from '../../../core/MmlTree/MmlNode.js';
+import {MmlMsubsup} from '../../../core/MmlTree/MmlNodes/msubsup.js';
+import {MmlMunderover} from '../../../core/MmlTree/MmlNodes/munderover.js';
+import {MmlMo} from '../../../core/MmlTree/MmlNodes/mo.js';
+import {Label} from '../Tags.js';
+
+
+// Namespace
+let BaseMethods: Record<string, ParseMethod> = {};
+
+// TODO: Once we can properly load AllEntities, there should be no need for
+//       those anymore.
```

Comment:
No, originally you had to instantiate an entities object form an entities class, but now there is just one entities object, and you simply load it and reference its properties.  You've done that elsewhere for the function-apply entity, as I recall.

Reply:
Yes, but when I use `Entities.ENTITIES.nbsp` for example it is undefined. So I still need to create it but there will only ever be one copy. In other words it's a singleton, correct?

Reply:
I think you want `Entities.entities.nbsp` (the case was changed somewhere along the way).

Reply:
Right, but that does not work here yet. Because that is in `master`, which is not merged into this PR.
I could merge `master` into `tex_input_typescript`, make the fix. That should make merging back into `master` easier.

Reply:
Yes, merging master in would probably make things easier.

Reply:
Done that. Also resolves the merge conflict with master.